pkg/reconciler/subscription/subscription.go

```diff
@@ -62,12 +64,17 @@ type Reconciler struct {
 	// DynamicClientSet allows us to configure pluggable Build objects
 	dynamicClientSet dynamic.Interface
 
+	// crdLister is used to resolve the ref version
+	crdLister apiextensionsv1lister.CustomResourceDefinitionLister
+
 	// listers index properties about resources
 	subscriptionLister  listers.SubscriptionLister
 	channelLister       listers.ChannelLister
 	channelableTracker  eventingduck.ListableTracker
 	destinationResolver *resolver.URIResolver
 	tracker             tracker.Interface
+
+	featureFlagsStore *feature.Store
```

Comment:
Maybe here too, if we had a context enricher type interface we can avoid leaking the details of how we look for feature flags and updating contexts.

Reply:
Solved